zfec/filefec.py

```diff
@@ -251,7 +253,7 @@ def decode_from_files(outf, infiles, verbose=False):
         m = nm
         if not (k is None or k == nk):
             raise CorruptedShareFilesError("Share files were corrupted -- share file %r said that k was %s but another share file previously said that k was %s" % (f.name, nk, k,))
-        if k > len(infiles):
+        if not (k is None or k <= len(infiles)):
```

Comment:
Good point. It's probably still worth putting this in a separate commit (which should appear before the general port-to-py3 commits), since it fixes a non-py3-specific problem (comparing None against an integer doesn't make sense in py2 either, even if it's not treated as an error)